### PR TITLE
convert image files to webp by default

### PR DESCRIPTION
**Pull request by xn**
This gets the idea going. When we add thumbnails, I'm sure this will turn into a Lifecycle.

Right now, we convert all images to webp. This can be overridden by using your own processor.

Minimum version of elixir supported is bumped to 1.13.0

### CODE REVIEW

**Review by TheFirstAvenger — APPROVED** [review]

**Comment by xn** [comment]
@leandrocp Which versions of Elixir are we targeting? Image is failing to compile on 1.12.0

**Comment by leandrocp** [comment]
> @leandrocp Which versions of Elixir are we targeting? Image is failing to compile on 1.12.0

1.12 but we can bump min that supports Image

File: test/support/fixtures/image.gif
Comment (leandrocp) [inline test/support/fixtures/image.gif:1]: Not a blocker but can we reduce the size of those support images?

File: lib/beacon/config.ex [hunk 1]
```diff
@@ -63,7 +63,10 @@ defmodule Beacon.Config do
   """
   @type media_type_config :: [
           {:backends, list(backend :: module())},
-          {:validations, list(validation_fun :: (Ecto.Changeset.t(), Beacon.Admin.MediaLibrary.FileMetadata.t() -> Ecto.Changeset.t()))}
+          {:validations, list(validation_fun :: (Ecto.Changeset.t(), Beacon.Admin.MediaLibrary.UploadMetadata.t() -> Ecto.Changeset.t()))},
+          {:processor,
+           {prossesor_fun ::
```
Comment (leandrocp) [inline lib/beacon/config.ex:68]: If I'm not wrong, I see this fn called with 1 arg. Is this spec correct?

File: mix.exs [hunk 2]
```diff
@@ -43,6 +43,7 @@ defmodule Beacon.MixProject do
       {:floki, ">= 0.30.0", only: :test},
       {:gettext, "~> 0.20"},
       {:heroicons, "~> 0.5"},
+      {:image, "~> 0.31.1"},
```
Comment (leandrocp) [inline mix.exs:46]:
```suggestion
      {:image, "~> 0.31"},
```

As a lib we can't be too restrictive :)

**Review by leandrocp — APPROVED** [review]

File: lib/beacon/config.ex (hunk 1 above)
Comment (xn) [reply lib/beacon/config.ex]: You are correct. I hadn't updated that to my final implementation. Great catch!

File: test/support/fixtures/image.gif
Comment (xn) [reply test/support/fixtures/image.gif:1]: is 100k fine or do you want tiny?

Comment (leandrocp) [reply test/support/fixtures/image.gif:1]: 100k is fine 👍🏻

**Pull request merged**